cmd/dao_tests/pubkey_resource_test.go

```diff
@@ -0,0 +1,137 @@
+//go:build integration
+// +build integration
+
+package main
+
+import (
+	"context"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/factories"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	_ "github.com/RHEnVision/provisioning-backend/internal/dao/sqlx"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func createPubkeyResourceNoop() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeNoop,
+		Handle:   factories.GetSequenceName("handle"),
+	}
+}
+
+func createPubkeyResourceAzure() *models.PubkeyResource {
+	return &models.PubkeyResource{
+		Tag:      "tag1",
+		PubkeyID: 1,
+		Provider: models.ProviderTypeAzure,
+		Handle:   factories.GetSequenceName("handle"),
+	}
+}
+
+func setupResource(t *testing.T) (dao.PubkeyResourceDao, context.Context) {
+	setup()
+	ctx := identity.WithTenant(t, context.Background())
+	resourceDao, err := dao.GetPubkeyResourceDao(ctx)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Just for the record: I just made a proposal on our Slack to actually move away from returning an error for all dao.GetXXX functions and simply panicking right there. This will make our life much easier.